Log deprecation warnings for Query class

tomalec opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Log deprecation warnings for Query class
deprecated in https://github.com/woocommerce/woocommerce/pull/49425

We introduced `GenericQuery` as a replacement for redundant `Query` classes, https://github.com/woocommerce/woocommerce/pull/49425 also added the support for Querry-less connection between a Controller and its DataStore.
As 9.2 is already prepared for the release with some developer advisory on how to 
With the new implementation, we can now start throwing warnings to motivate the community to stop using the classes so we can eventually remove them.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install an extension that uses the legacy `Query` class, like
   - woocommerce-product-bundles
   - woocommerce-gift-cards
   - sales-report-email-pro
3. Open the page that uses it (analytic reports)
4. Check the PHP log, look for `deprecated since version 9.3.0`


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Log deprecation warnings for `Query` class usage.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

tomalec committed:
Log deprecation warnings for Query class

deprecated in 0322426dce720e129e3804d6fad196c07e2489b3

tomalec committed:
Add a full stop at the end of code comment

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

ianlin commented on plugins/woocommerce/changelog/50969-dev-deprecate-query:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: tweak
+
+Log deprecation warnings for `Query` class usage.
```
I think it could be more specific to say that we're deprecating analytic reports Query class.

ianlin reviewed (approved):
Thanks for adding the deprecation warnings for the Analytics Reports Query class. I can see the warnings appearing in the debug.log when accessing the analytics pages for "Product Bundles" and "Gift Cards." I’ve left a comment suggesting the changelog could be more specific, but I’m approving it for now as I don't think it needs a second review.

tomalec committed:
Specify the deprecated class

Address https://github.com/woocommerce/woocommerce/pull/50969#discussion_r1732055959

Commit:
Add changefile(s) from automation for the following project(s): woocommerce
